hasher-matcher-actioner/webapp/src/settings/ThreatExchangeSettingsTab.jsx

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ */
+
+import React, {useEffect, useState} from 'react';
+import {
+  Col,
+  Row,
+  Button,
+  Toast,
+  Form,
+  Spinner,
+  Tooltip,
+  OverlayTrigger,
+  Card,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import {CopyableTextField} from '../utils/TextFieldsUtils';
+import {
+  fetchAllDatasets,
+  syncAllDatasets,
+  udpateDataset,
+  deleteDataset,
+} from '../Api';
+
+export default function ThreatExchangeSettingsTab() {
+  const [datasets, setDatasets] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [syncing, setSyncing] = useState(true);
+  const [showToast, setShowToast] = useState(false);
+  const [toastBody, setToastBody] = useState(null);
+  const onPrivacyGroupSave = privacyGroup => {
+    udpateDataset(
+      privacyGroup.privacyGroupId,
+      privacyGroup.localFetcherActive,
+      privacyGroup.localWriteBack,
+    ).then(response => {
+      datasets[
+        datasets.findIndex(
+          item => item.privacy_group_id === response.privacy_group_id,
+        )
+      ] = response;
+      setDatasets(datasets);
+    });
+  };
+  const onPrivacyGroupDelete = privacyGroupId => {
+    deleteDataset(privacyGroupId).then(() => {
+      const filteredDatasets = datasets.filter(
+        item => item.privacy_group_id !== privacyGroupId,
+      );
+      setDatasets(filteredDatasets);
+    });
+  };
+  const onSync = () => {
+    syncAllDatasets().then(syncResponse => {
+      setSyncing(true);
+      if (syncResponse.response === 'Dataset is update-to-date') {
```

Comment:
Just use the status code? The response string is rarely part of the contract and introduces an implicit dependency.

Reply:
IMO, if you get a 'then' call, you can assume the API call succeeded. If it were to fail, the `catch` method would be called.

The pattern I have seen most often: (I am biased. Not an expert)
```javascript
const onSaveHandler = () => {
  makeAPICall()
    .then(() => {
       ...updateState()
       ...successMessage()
    })
    .catch(() => {
       ...resetState()
       ...failedMessage()
    })

}

```